### ISSUE TITLE
linux cataclysm --version and main menu report wrong version in 0.G release

### ISSUE THREAD

**Issue by AshenColors** [CleverRaven/Cataclysm-DDA#73350]
### Describe the bug

The Linux build artifacts from the [0.G release page](https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/0.G), and the Fedora package downstream (and probably other distros) generated from this build, reports that the version is 0.F instead of 0.G, sending me down a rabbit hole to figure out why.

It looks like this was fixed in the *branch* 0.G-branch with [this commit on March 2nd, 2023](https://github.com/CleverRaven/Cataclysm-DDA/commit/8156eb3b65a9f79864c9e052980ece16a2d25d5a), but that's two commits ahead of the *tag* 0.G, which is what the release is based on.

### Attach save file

N/A, happens on first run and every subsequent run

### Steps to reproduce

```
ashen@slateblue:~$ sudo dnf install cataclysm-dda
[sudo] password for ashen: 
Last metadata expiration check: 0:32:45 ago on Sat 27 Apr 2024 09:54:46 PM CDT.
Dependencies resolved.
========================================================================================
 Package                         Architecture  Version              Repository     Size
========================================================================================
Installing:
 cataclysm-dda                   x86_64        0.G-1.fc39           fedora         12 M
Installing dependencies:
 SDL2_mixer                      x86_64        2.6.3-4.fc40         fedora        159 k
 SDL2_ttf                        x86_64        2.22.0-2.fc40        fedora         43 k
 cataclysm-dda-data              noarch        0.G-1.fc39           fedora         15 M
 cataclysm-dda-tiles-data        noarch        0.G-1.fc39           fedora         31 M
Installing weak dependencies:
 cataclysm-dda-tiles             x86_64        0.G-1.fc39           fedora         10 M

Transaction Summary
========================================================================================
Install  6 Packages

Total download size: 68 M
Installed size: 199 M
Is this ok [y/N]: y
Downloading Packages:
(1/6): SDL2_ttf-2.22.0-2.fc40.x86_64.rpm                298 kB/s |  43 kB     00:00    
(2/6): SDL2_mixer-2.6.3-4.fc40.x86_64.rpm               781 kB/s | 159 kB     00:00    
(3/6): cataclysm-dda-0.G-1.fc39.x86_64.rpm              4.7 MB/s |  12 MB     00:02    
(4/6): cataclysm-dda-tiles-0.G-1.fc39.x86_64.rpm        3.4 MB/s |  10 MB     00:03    
(5/6): cataclysm-dda-data-0.G-1.fc39.noarch.rpm         3.0 MB/s |  15 MB     00:04    
(6/6): cataclysm-dda-tiles-data-0.G-1.fc39.noarch.rpm   6.8 MB/s |  31 MB     00:04    
----------------------------------------------------------------------------------------
Total                                                   9.4 MB/s |  68 MB     00:07     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                1/1 
  Installing       : SDL2_ttf-2.22.0-2.fc40.x86_64                                  1/6 
  Installing       : SDL2_mixer-2.6.3-4.fc40.x86_64                                 2/6 
  Installing       : cataclysm-dda-data-0.G-1.fc39.noarch                           3/6 
  Installing       : cataclysm-dda-0.G-1.fc39.x86_64                                4/6 
  Installing       : cataclysm-dda-tiles-data-0.G-1.fc39.noarch                     5/6 
  Installing       : cataclysm-dda-tiles-0.G-1.fc39.x86_64                          6/6 
  Running scriptlet: cataclysm-dda-tiles-0.G-1.fc39.x86_64                          6/6 

Installed:
  SDL2_mixer-2.6.3-4.fc40.x86_64           SDL2_ttf-2.22.0-2.fc40.x86_64                
  cataclysm-dda-0.G-1.fc39.x86_64          cataclysm-dda-data-0.G-1.fc39.noarch         
  cataclysm-dda-tiles-0.G-1.fc39.x86_64    cataclysm-dda-tiles-data-0.G-1.fc39.noarch   

Complete!
ashen@slateblue:~$ cataclysm --version
Cataclysm Dark Days Ahead: 0.F

-tiles, -sound

data dir: /usr/share/cataclysm-dda/
user dir: /home/ashen/.cataclysm-dda/
ashen@slateblue:~$ cd Downloads/
ashen@slateblue:~/Downloads$ wget https://github.com/CleverRaven/Cataclysm-DDA/releases/download/0.G/cdda-linux-curses-x64-2023-03-01-0054.tar.gz
cdda-linux-curses-x6 100% [=====================================>]   63.74M   10.79MB/s
                          [Files: 1  Bytes: 63.74M [8.78MB/s] Red]
ashen@slateblue:~/Downloads$ tar --exclude="*/*" -tf cdda-linux-curses-x64-2023-03-01-0054.tar.gz 
cataclysmdda-0.F/
ashen@slateblue:~/Downloads$ 
```

### Expected behavior

The version folder in the release artifact, the executable's --version flag, and the main menu should display the correct release version of 0.G.

### Screenshots

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/3249074/db4591a8-885f-4b2a-a4aa-4e6f537caf41)

### Versions and configuration

- OS: Linux
    - OS Version: <unknown>
- Game Version: 0.F [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by NetSysFire** [comment]
This is valid and all but 0.G will not receive further backports, especially since 0.H is quite close. Plus this is more or less a cosmetic issue which is not catastrophic and thus not in particular need of backporting. It was fixed in #70663 for stable and in #70664 for experimental.

Thanks for the thorough report though.

**Comment by AshenColors** [comment]
> This is valid and all but 0.G will not receive further backports, especially since 0.H is quite close. Plus this is more or less a cosmetic issue which is not catastrophic and thus not in particular need of backporting. It was fixed in #70663 for stable and in #70664 for experimental.
> 
> Thanks for the thorough report though.

I agree a backport is unnecessary, but could I suggest editing the Github release description text to add that "yes,, the linux folder/file says 0.F but it's actually 0.G, trust us"? I don't think that would require an actual new build or trigger any automated workflows.

Since this looks like a downstream packaging issue at first glance, it would save whoever notices the same thing a few hours tracing the packaging and build pipeline for the apparent issue.

In any case, I appreciate the acknowledgement and look forward to being just as hopelessly but enjoyably overwhelmed with 0.H.

**Comment by I-am-Erk** [comment]
There were some versioning issues with 0.G but it is, indeed, just late for us to bother fixing them. We're a matter of weeks from H.
